R/input-check-helpers.R

```diff
@@ -195,19 +195,59 @@ check_equal_length <- function(...,
 #' @keywords internal
 
 check_metrics <- function(metrics) {
-  # use all available metrics if none are given
+
+  # use all available metrics if no metrics are provided at all and return
+  available_metrics <- available_metrics()
   if (is.null(metrics)) {
-    metrics <- available_metrics()
+    metrics <- available_metrics
```

Comment:
why not pass in a list of metrics (as a list name pair) with this as the default argument?